SmartDeviceLinkTests/SDLCacheFileManagerSpec.m

```diff
@@ -0,0 +1,313 @@
+
+                it(@"should return retrieved image and no error", ^{
+                    expect(resultImage).to(equal(testImage));
+                    expect(resultError).to(beNil());
+                });
+            });
+        });
+
+        context(@"request with expired icon", ^{
+            __block UIImage *resultImage = nil;
+            __block NSError *resultError = nil;
+
+            context(@"image download succeeds", ^{
+                context(@"Failed to update archive object", ^{
+                    beforeEach(^{
+                        OCMStub(ClassMethod([mockArchiver archiveRootObject:[OCMArg any] toFile:[OCMArg any]])).andReturn(NO);
+                        OCMStub(ClassMethod([testManagerMock sdl_writeImage:[OCMArg any] toFileFromURL:[OCMArg any] atFilePath:[OCMArg any]])).andReturn(testFilePath);
+                        OCMStub([testManagerMock sdl_downloadIconFromRequestURL:[OCMArg any] withCompletionHandler:([OCMArg invokeBlockWithArgs:testImage, [NSNull null], nil])]);
+
+                        [testManager retrieveImageForRequest:expiredTestRequest withCompletionHandler:^(UIImage * _Nullable image, NSError * _Nullable error) {
+                            resultImage = image;
+                            resultError = error;
+                        }];
+                    });
+
+                    it(@"it should return downloaded image and no error", ^{
+                        expect(resultImage).to(equal(testImage));
+                        expect(resultError).to(beNil());
+                    });
+                });
+
+                context(@"write image to file path succeeds", ^{
+                    beforeEach(^{
+                        OCMStub(ClassMethod([mockArchiver archiveRootObject:[OCMArg any] toFile:[OCMArg any]])).andReturn(YES);
+                        OCMStub(ClassMethod([testManagerMock sdl_writeImage:[OCMArg any] toFileFromURL:[OCMArg any] atFilePath:[OCMArg any]])).andReturn(testFilePath);
+                        OCMStub([testManagerMock sdl_downloadIconFromRequestURL:[OCMArg any] withCompletionHandler:([OCMArg invokeBlockWithArgs:testImage, [NSNull null], nil])]);
+
+                        [testManager retrieveImageForRequest:expiredTestRequest withCompletionHandler:^(UIImage * _Nullable image, NSError * _Nullable error) {
+                            resultImage = image;
+                            resultError = error;
+                        }];
+                    });
+
+                    it(@"it should return downloaded image and no error", ^{
+                        expect(resultImage).to(equal(testImage));
+                        expect(resultError).to(beNil());
+                    });
+                });
+
+                context(@"write image to file path fails", ^{
+                    beforeEach(^{
+                        OCMStub(ClassMethod([testManagerMock sdl_writeImage:[OCMArg any] toFileFromURL:[OCMArg any] atFilePath:[OCMArg any]])).andReturn(nil);
```

Comment:
Shouldn't this stub

```
OCMStub(ClassMethod([mockArchiver archiveRootObject:[OCMArg any] toFile:[OCMArg any]])).andReturn(YES);
```

Reply:
If this method fails to write the image to file, it will return nil and we return `completion(image, nil)`.  My thoughts were `archiveRootObject:ToFile:` should not be called in this instance because `updateArchiveFileWithIconURL:iconFilePath:archiveFile:error` should not be called without a file path.